server/me.go

```diff
@@ -271,17 +271,39 @@ func (s *Service) Me(w http.ResponseWriter, r *http.Request) {
 		SuperAdmin: s.newUsersAreSuperAdmin(),
 	}
 
+	superAdmin := s.mapPrincipalToSuperAdmin(p)
+	if superAdmin {
+		user.SuperAdmin = superAdmin
+	}
+
 	roles, err := s.mapPrincipalToRoles(serverCtx, p)
 	if err != nil {
 		Error(w, http.StatusInternalServerError, err.Error(), s.Logger)
 		return
 	}
 
-	if len(roles) == 0 {
+	if !superAdmin && len(roles) == 0 {
```

Comment:
As a general comment, reasoning about this function is hard because it handles both new user creation as well as existing user validation.